monitor/api/tests/tests.py

```diff
@@ -0,0 +1,163 @@
+import json
+from typing import Any, Dict, List, Optional
+
+from clients.api.api import APIClient
+from monitor.api.tests.schema import InvocationsSchems, TestMetadataSchema, TestUniqueIdType, InvocationSchema, TotalsInvocationsSchema
+
+
+class TestsAPI(APIClient):
+    @staticmethod
+    def _get_test_sub_type_unique_id(test: TestMetadataSchema) -> str:
+        return f"{test.test_unique_id}.{test.test_type}.{test.test_sub_type if test.test_sub_type else 'None'}.{test.table_name if test.table_name else 'None'}.{test.column_name if test.column_name else 'None'}"
+
+    def get_tests_metadata(self, days_back: Optional[int] = 7) -> List[TestMetadataSchema]:
+        tests_metadata = json.loads(self.dbt_runner.run_operation(macro_name='get_test_results', macro_args=dict(days_back=days_back))[0])
+        return [TestMetadataSchema(**test_metadata) for test_metadata in tests_metadata]
+
+    def get_metrics(
+        self,
+        tests_metadata: Optional[List[TestMetadataSchema]] = None,
+        days_back: Optional[int] = None,
+        metrics_sample_limit: int = 5
+    ) -> Dict[TestUniqueIdType, Dict[str, Any]]:
+        tests: List[TestMetadataSchema] = tests_metadata if tests_metadata is not None else self.get_tests_metadata(days_back=days_back)
+        tests = [dict(test) for test in tests]
+        return json.loads(self.dbt_runner.run_operation(
+            macro_name='get_tests_metrics',
+            macro_args=dict(tests=tests, metrics_sample_limit=metrics_sample_limit)
```

Comment:
This could be an issue actually - CLI has a limit in terms of the length of the params (this is why we split the slack ids to batches). Let's discuss about it?